ortools/linear_solver/xpress_interface.cc

```diff
@@ -841,8 +841,7 @@ void XpressInterface::SetVariableInteger(int var_index, bool integer) {
   // (supportIncrementalExtraction is true) then we MUST change the
   // type of extracted variables here.
 
-  if (!supportIncrementalExtraction && !slowUpdates &&
-      !SlowSetVariableInteger) {
+  if (!supportIncrementalExtraction && !(slowUpdates & SlowSetVariableInteger)) {
```

Comment:
this is not equivalent to before, why did you have to change it ?

Reply:
Because the previous version contained a typo, see https://c-faq.com/misc/bitsets.html for bitset doc.

Reply:
i undesrtant that we should have used & instead of &&. but since you factorized the ! condition, don't you want to use | instead of & ?

Reply:
- `!supportIncrementalExtraction` is `true`, both before and after the fix.
- `slowUpdates` is `SlowSetObjectiveCoefficient | SlowClearObjective`
- Before the fix, `!slowUpdates` is `false`, so `!supportIncrementalExtraction && !slowUpdates && !SlowSetVariableInteger` is also `false`
- After the fix, we test whether the bit `SlowSetVariableInteger` is set to 1

You can see at l. 713, 840, 903, 936, 977, 998 in **xpress_interface.cc**.